core/domain/translation_services.py

```diff
@@ -85,3 +90,207 @@ def get_and_cache_machine_translation(
         )
 
     return translated_text
+
+
+def add_new_translation(
+    entity_type: feconf.TranslatableEntityType,
+    entity_id: str,
+    entity_version: int,
+    language_code: str,
+    content_id: str,
+    translated_content: translation_domain.TranslatedContent
+) -> None:
+    """Adds new translated content for the entity in the EntityTranslation
+    model.
+
+    Args:
+        entity_type: TranslatableEntityType. The type of the entity.
+        entity_id: str. The ID of the entity.
+        entity_version: int. The version of the entity.
+        language_code: str. The language code for the entity.
+        content_id: str. The Id of the content.
+        translated_content: TranslatedContent. The translated content object.
+    """
+
+    entity_translation = translation_fetchers.get_entity_translation(
+        entity_type, entity_id, entity_version, language_code)
+
+    entity_translation.translations[content_id] = translated_content
+
+    entity_translation.validate()
+
+    model = translation_models.EntityTranslationsModel.create_new(
+        entity_type.value,
+        entity_id,
+        entity_version,
+        language_code,
+        entity_translation.to_dict()['translations']
+    )
+
+    model.update_timestamps()
```

Comment:
A bit too much empty lines.

Reply:
They were different steps i.e, fetch, update, validate, create and put. I've changed it to update and put section!